src/math/warpExposure.cc

```diff
@@ -494,6 +496,228 @@ int warpImage(DestImageT &destImage, SrcImageT const &srcImage, geom::XYTransfor
+                    // and positions for the previous row for larger indices (including endCol)
+                    geom::Point2D leftSrcPos = srcPosView[prevEndCol];
+                    geom::Point2D rightSrcPos = srcPosView[endCol] + yDeltaSrcPosList[colBand];
+                    geom::Extent2D xDeltaSrcPos = (rightSrcPos - leftSrcPos) * invWidthList[colBand];
+
+                    for (int col = prevEndCol + 1; col <= endCol; ++col, ++destXIter) {
+                        geom::Point2D leftSrcPos = srcPosView[col - 1];
+                        geom::Point2D srcPos = leftSrcPos + xDeltaSrcPos;
+                        double relativeArea = computeRelativeArea(srcPos, leftSrcPos, srcPosView[col]);
+
+                        srcPosView[col] = srcPos;
+
+                        if (warpAtOnePoint(
+                                    destXIter, srcPos, relativeArea,
+                                    typename image::detail::image_traits<DestImageT>::image_category())) {
+                            ++numGoodPixels;
+                        }
+                    }  // for col
+                }      // for col band
+            }          // for row
+        }              // while next row band
+
+    } else {
+        // No interpolation
+
+        // prevSrcPosList = source positions from the previous row; these are used to compute pixel area;
+        // to begin, compute sources positions corresponding to destination row = -1
+        std::vector<geom::Point2D> destPosList;
+        destPosList.reserve(1 + destWidth);
+        for (int col = -1; col < destWidth; ++col) {
+            destPosList.emplace_back(geom::Point2D(col, -1));
+        }
+        auto prevSrcPosList = localDestToSrc.tranForward(destPosList);
+
+        for (int row = 0; row < destHeight; ++row) {
+            destPosList.clear();
+            for (int col = -1; col < destWidth; ++col) {
+                destPosList.emplace_back(geom::Point2D(col, row));
+            }
+            auto srcPosList = localDestToSrc.tranForward(destPosList);
+
+            typename DestImageT::x_iterator destXIter = destImage.row_begin(row);
+            for (int col = 0; col < destWidth; ++col, ++destXIter) {
+                // column index = column + 1 because the first entry in srcPosList is for column -1
+                auto srcPos = srcPosList[col + 1];
+                double relativeArea =
+                        computeRelativeArea(srcPos, prevSrcPosList[col], prevSrcPosList[col + 1]);
+
+                if (warpAtOnePoint(destXIter, srcPos, relativeArea,
+                                   typename image::detail::image_traits<DestImageT>::image_category())) {
+                    ++numGoodPixels;
+                }
+            }  // for col
+            // move points from srcPosList to prevSrcPosList (we don't care about what ends up in srcPosList
+            // because it will be reallocated anyway)
+            std::swap(srcPosList, prevSrcPosList);
```

Comment:
OK. I hate how that makes it hard to tell where `swap` is defined, but it does appear to be standard.